### PR TITLE
Chore: used 'enhanced' for loop, and discussed dependency management.

### PR DESCRIPTION
**Pull request by dbmalkovsky**
There are 31 occurrences of using 'enhanced' for loop; this PR makes the remaining 5 match the others.

### CODE REVIEW

**Comment by BlakeTNC** [comment]
@dbmalkovsky 
Question: are you using some kind of automated code improvement tool to do some or all of these cleanup chores?

I ask because I noted that some of these edits are going inside of the "privatejgoodies" folder. That folder contains a private copy of an external dependency. I don't remember why I included the dependency code directly instead of a maven dependency or a jar. I'm guessing I had a reason. But anyways, I have never historically edited anything in that folder that I remember, and doing so makes me uneasy. Because: If we accidentally introduce a bug in that folder, it would be very hard, or impractically difficult for me or likely anyone else to debug.

If you are using an automated cleanup tool, then I imagine there is little chance of introducing a bug there. But if you are doing manual edits, I imagine subtle mistakes could happen.

PS, I think I remember the reason for that kind of import. It may have been so that people who use this library, could also add their own versions of jgoodies to their projects without any dependency collisions.

**Comment by BlakeTNC** [comment]
@WiseEagleOwl Please let me know if you have any thoughts on my comment. Thanks.

**Comment by BlakeTNC** [comment]
@WiseEagleOwl 
I looked at the pom and it occurs to me that we added another dependency (junit) in pull #81. I wonder if this could cause a similar problem for users in large projects as I mentioned above, dependency collisions. 

Here is a description of the problem. https://www.baeldung.com/maven-version-collision

It looks like I originally designed this project to have few or no external dependencies for this reason. And I included jgoodies directly in the code with a renamed package for the same reason. (jgoodies was a non-optional dependency.)

Since junit is likely optional, I wonder if we should do something about that item as well. Options might be:
- To exclusively use built-in java testing functionality (assert, etc), if that exists, or 
- I wonder if the junit dependency and test classes could somehow be removed while packaging releases?
- Or there might be other options that I don't know about, to mitigate the potential problem.

**Comment by BlakeTNC** [comment]
I looked up how others handle test dependency removal. They say unit testing should ideally be in a separate project. That way the main project has no need for the unit test classes or the testing library dependency.  Although some of these answers are for C#, the same concept could apply in Java. 

References:
https://stackoverflow.com/questions/7258404/remove-nunit-reference-for-release-build
https://stackoverflow.com/questions/1278831/unit-tests-in-production-release-code

**Comment by dbmalkovsky** [comment]
@BlakeTNC Yes I use a series of tools to find and "correct" issues.  One is the Analyze tool that comes with IntelliJ.  I also use SonarLint, PMD, and Spotbugs.

**Comment by WiseEagleOwl** [comment]
@BlakeTNC the project is using `maven-shade-plugin` to remove all public dependencies by setting the public POM to `pom-for-users.tx`. It has been like this since I started contributing and I did not change that workflow. Consumers of the maven release only see an empty dependencies section:
 https://github.com/LGoodDatePicker/LGoodDatePicker/blob/cad54eaf550a6ec55298d710366017f18e8d9c71/Project/src/main/resources/pom-for-users.txt#L38-L39
This means no consumer sees our internal dependency to `junit`. Having the tests in the project itself is very useful to assure that programmers who change the library code execute them automatically on build and see possible test failures. 

The test classes are also not included in the released JAR files, which can be verified by renaming the jar files to ZIP files and opening it.
If we had any accidental dependencies that were actually needed by the *LGoodDatePicker* library itself then there would be ` ClassNotFoundException`s because of the shading nobody knows where to find them.

In conclusion: All current *LGoodDatePicker* releases have no dependencies. The Project itself has them because they are needed for development (no one should develop the project without running the tests).

If you have any other questions or concerns regarding this just let me know.

**Comment by WiseEagleOwl** [comment]
@BlakeTNC regarding changes in the `privatejgoodies` folder: my bottleneck is the time I can donate to review changes. 
As far as I understand it our only dependency to that code is the `FormLayout` used for the `CalendarPanel` and `DatePicker` components.
Due to my time constraints any further development in that part of the code is out of scope to me. I personally would only fix bugs affecting the `FormLayout` or compiler warnings in that folder (to keep the project warnings to zero).
If at some point in the future `privatejgoodies` gets unusable due to changes in the JDK or what else might happen I would prefer to move away from `FormLayout` instead of investing the time to refactor and fix it. 
If someone were to donate the code for a working `FormLayout` replacement I would even vote to replace `FormLayout` rather sooner than later to reduce the amount of code we have to maintain.

**Comment by BlakeTNC** [comment]
@WiseEagleOwl 
> the project is using maven-shade-plugin to remove all public dependencies by setting the public POM to pom-for-users.txt

Thanks for pointing that out. That's great. I feel like thats something I should have known, lol. (Since I wrote it.) I updated the pom docs to emphasize that a simpler pom is used for releases, and that JUnit is not a release dependency. 8f93620ea7ba55df16dca8fd750be1cddcc1208d

> If someone were to donate the code for a working FormLayout replacement I would even vote to replace FormLayout rather sooner than later to reduce the amount of code we have to maintain.

I agree. I will create an issue for this, as an enhancement. 
I looked into possible form replacements. GridBagLayout may be an option, and that is built into java. MigLayout may be an option. It has fewer classes and folders than JGoodies, and is still actively maintained. It is offered in source code form. So if we used it we could do the same package rename trick to avoid dependency collisions. https://github.com/mikaelgrev/miglayout

Between the two, I'd vote for GridBagLayout if it is at all capable of doing the job, because no new classes in our source. 

@dbmalkovsky 
> Yes I use a series of tools to find and "correct" issues.

Ok, thats nice. It'g great that you are careful. Also, it's fine to make changes in the jgoodies folder for completeness if your cleanup process. I realized that if jgoodies were to be broken, we could just unzip the backup to fix it. The backup is here. 
https://github.com/LGoodDatePicker/LGoodDatePicker/blob/master/zExtraFiles/CompleteWorkingSourcePrivateJGoodiesForms1.7.2%2C%2017mar16.zip

**Comment by BlakeTNC** [comment]
A new enhancement issue for replacing the layout manager was created, at #168.
